src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1219,6 +1224,43 @@ void OrbitMainWindow::on_actionOpen_Capture_triggered() {
   QProcess::startDetached(orbit_executable, arguments << file << command_line_flags_);
 }
 
+void OrbitMainWindow::on_actionRename_Capture_File_triggered() {
+  CHECK(target_label_->GetFilePath().has_value());
+  const std::filesystem::path& current_file_path = target_label_->GetFilePath().value();
+  QString file_path =
+      QFileDialog::getSaveFileName(this, "Rename or Move capture...",
+                                   QString::fromStdString(current_file_path.string()), "*.orbit");
+
+  std::filesystem::path new_file_path{file_path.toStdString()};
+
+  if (new_file_path == current_file_path) return;
+
+  auto* progress_dialog = new QProgressDialog(
+      QString("Moving file to \"%1\"...").arg(QString::fromStdString(new_file_path.string())), "",
+      0, 0, this);
+  progress_dialog->setWindowModality(Qt::WindowModal);
+  progress_dialog->show();
+
+  orbit_base::Future<ErrorMessageOr<void>> rename_future =
+      app_->MoveCaptureFile(current_file_path, new_file_path);
+
+  rename_future.Then(main_thread_executor_.get(), [this, progress_dialog, current_file_path,
+                                                   new_file_path](ErrorMessageOr<void> result) {
+    progress_dialog->close();
+    if (result.has_error()) {
+      QMessageBox::critical(
+          this, QString::fromStdString("Unable to Rename File"),
+          QString::fromStdString(absl::StrFormat(R"(Unable to rename/move file "%s" -> "%s": %s)",
+                                                 current_file_path.string(), new_file_path.string(),
+                                                 result.error().message())));
+      return;
+    }
+
+    target_label_->SetFile(new_file_path);
+    setWindowTitle(QString::fromStdString(new_file_path.string()));
```

Comment:
My first instinct when reading this: Shouldn't there be a common functionality to update the UI when the filename has changed that is also used when opening an existing file?

Reply:
let me do a follow up CL for this. seems like this belongs to a different commit.